updating future projects page (#265)
ros-planning/moveit.ros.org: documentation/contributing/future_projects/index.markdown

<img src="/assets/images/logo/moveit_logo-black.png" width="300"/>

> This page lists potential projects that would greatly benefit the MoveIt! project. This is by no means an exhaustive list, but is meant to serve as a discussion starting point for code sprints and other efforts. Please contact PickNik Consulting (hello@picknik.ai) for further information or sponsorship opportunities.

Reviewer: ```suggestion
This page lists potential projects that would greatly benefit the MoveIt! project. This is by no means an exhaustive list, but is meant to serve as a discussion starting point for code sprints and other efforts. Please contact [PickNik Consulting](https://picknik.ai/connect/) for further information or sponsorship opportunities.
```

## TrajOpt Integration

> TrajOpt was recently created at UC Berkeley as motion planning framework for generating robot trajectories by local optimization. Integration of TrajOpt into the MoveIt! framework would require API changes to how MoveIt! stores and accesses information about the kinematic chain and changes to the MoveIt! collision checking API to support integration with the Bullet collision checker for convex to convext contact checking.

Reviewer: ```suggestion
TrajOpt is a motion planning framework from UC Berkeley for generating robot trajectories by local optimization. Integration of TrajOpt into the MoveIt! framework would require API changes to how MoveIt! stores and accesses information about the kinematic chain and changes to the MoveIt! collision checking API to support integration with the Bullet collision checker for convex to convex contact checking.
```

2013 wasn't that recent

## Perception Pipeline Improvements

MoveIt! is not a perception library, but does come with support for using octomaps with point clouds. This functionality has not been well maintained and needs extensive improvements:

- Improve speed of plugins, fix locking issues
- Create tutorials and documentation for using this functionality
- Add support for object segmentation, table detection, etc
- Possible integration of Neural Networks for image recognition

## Improve MoveIt! Grasps Library

MoveIt! has long lacked a good grasping library for generating candidate object grasps for manipulation. With the release of [MoveIt! Grasps] (https://github.com/PickNikRobotics/moveit_grasps/), we are working to fill this gap. Although MoveIt! Grasps is a big step up from the pick-and-place pipeline, it is far from perfect. Improvements to MoveIt! grasps could include:

  - Integration with a machine-learning-based grasp generator and/or grasp scoring package such as [Agile Grasps](http://wiki.ros.org/agile_grasp)
  - Improve tutorials and documentation for grasping in MoveIt!
  - Improve test coverage, especially for grasp generation

## Cartesian Planning improvements

> The default cartesian planner in MoveIt! is extremely naive and has enormous room for improvement. Improvements to the cartesian planner could include:

Reviewer: ```suggestion
The default cartesian planner in MoveIt! is naive and has enormous room for improvement. Improvements to the Cartesian planner could include:
```
